pkg/controller/common/helpers.go

```diff
@@ -22,3 +25,17 @@ func WriteTerminationError(err error) {
 	ioutil.WriteFile("/dev/termination-log", []byte(msg), 0644)
 	glog.Fatal(msg)
 }
+
+// ValidateIgnition evaluates whether a given non-empty Ignition Config is valid
```

Comment:
How about:

```
// ValidateIgnition wraps the underlying Ignition validation, but explicitly supports
// a completely "empty" Ignition config as valid in addition.  This is because we
// want to allow MachineConfig objects which just have e.g. KernelArguments
// set, but no Ignition config.
// Returns nil if the config is valid (per above) or an error otherwise.
```

Reply:
this is much better. thanks!

Reply:
done